core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -248,6 +247,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
 
     initializeControllerContext()
 
+    val pendingPreferredReplicaElections = fetchPendingPreferredReplicaElections()
+    val (topicsToBeDeleted, topicsIneligibleForDeletion) = fetchTopicDeletionsInProgress(pendingPreferredReplicaElections)
```

Comment:
I think this probably needs to be computed after maybeTriggerPartitionReassignment(), which may complete some of the partition reassignment.